addons/mail/controllers/discuss.py

```diff
@@ -285,3 +285,65 @@ def message_get_suggested_recipients(self, model, res_ids):
         except Exception:
             return {}
         return records._message_get_suggested_recipients()
+
+    # --------------------------------------------------------------------------
+    # RTC API
+    # --------------------------------------------------------------------------
+
+    @http.route('/mail/rtc/session/notify_call_members', type="json", auth="public")
+    def session_call_notify(self, sender, targets=None, content=None):
+        if request.env.user._is_public():
+            guest = request.env['mail.guest']._get_guest_from_request(request)
+            if guest:
+                guest.env['mail.channel.rtc.session'].sudo().browse(int(sender))._notify_peers(targets, content)
+                return
+        request.env['mail.channel.rtc.session'].sudo().browse(int(sender))._notify_peers(targets, content)
+
+    @http.route('/mail/rtc/session/update_and_broadcast', type="json", auth="public")
+    def session_update_and_broadcast(self, session_id, values):
+        if request.env.user._is_public():
+            guest = request.env['mail.guest']._get_guest_from_request(request)
+            if guest:
+                guest.env['mail.channel.rtc.session'].sudo().browse(int(session_id))._update_and_broadcast(values)
+                return
+        request.env['mail.channel.rtc.session'].sudo().browse(int(session_id))._update_and_broadcast(values)
+
+    @http.route('/mail/rtc/channel/join_call', type="json", auth="public")
+    def channel_call_join(self, channel_id):
+        if request.env.user._is_public():
+            guest = request.env['mail.guest']._get_guest_from_request(request)
+            if guest:
+                return guest.env['mail.channel'].sudo().browse(int(channel_id))._join_call()
```

Comment:
When you have the `channel_id` it's better to simply use `channel_partner_sudo = request.env['mail.channel.partner']._get_as_sudo_from_request_or_raise(request=request, channel_id=int(channel_id))` like other routes.